src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/ProgramStageSelector/ProgramStageSelector.container.tsx

```diff
@@ -106,11 +109,13 @@ export const ProgramStageSelector = ({ programId, orgUnitId, teiId, enrollmentId
                     header={i18n.t('Choose a stage for a new event')}
                     noncollapsible
                 >
-                    <ProgramStageSelectorComponent
-                        programStages={availableStages}
-                        onSelectProgramStage={onSelectProgramStage}
-                        onCancel={onCancel}
-                    />
+                    {rulesReady ?
+                        <ProgramStageSelectorComponent
+                            programStages={availableStages}
+                            onSelectProgramStage={onSelectProgramStage}
+                            onCancel={onCancel}
+                        />
+                        : <LoadingMaskElementCenter />}
```

Comment:
I'm wondering if using `LoadingMaskElementCenter` is the most sensible approach because the ticket mentions exploring a solution that will not block the UI while the rules engine is running. Is there a more appropriate UX pattern to indicate to the user that "some work is happening in the background" and, at the same time, display the DataEntry forms? Perhaps the design team has some UX suggestions.